### ISSUE TITLE
[1.12.2] Smoke particles can leak into other dimensions.

### ISSUE THREAD

**Issue by SonicX8000** [jbredwards/Campfire-Mod#7]
Minecraft: 1.12.2
Forge: 14.23.5.2860

Mods: 4
Campfire-v1.1.3-mc1.12.2
HadEnoughItems_1.12.2-4.24.5
Xaeros_Minimap_23.3.0_Forge_1.12
XaerosWorldMap_1.29.2_Forge_1.12

---

If a chunk is loaded and there is a brazier or campfire in that chunk... the smoke particles will leak into other dimensions at the same location and remain visible until the chunk that has them unloads. This can be a problem if one is using chunk loaders to keep chunks loaded at all times.

![image](https://user-images.githubusercontent.com/36653806/229349159-7c1137ae-69b5-45e5-b233-86776bbba12b.png)

For this image I am at this spot in the overworld with the listed coordinates. I shall then enter the nether and teleport to the same coordinates.

![image](https://user-images.githubusercontent.com/36653806/229349185-888b30e8-2b52-4fb1-a0e0-e2c94aabb8ea.png)

I teleported to the same coordinates and I can see the smoke particles rendering... though after awhile they stop rendering prob due to the chunk unloading.

It looks like it may only be just the braziers but it wouldn't hurt to see if campfires may somehow leak into other dimensions as well. The minimap mod is not needed as you can get the coordinates via F3 but I figure it might make it easier to showcase the bug.

**Comment by jbredwards** [comment]
I can't recreate this issue, but does the attached build fix it? (remember to unzip, github doesn't support .jar files)
[Campfire-v1.1.4-mc1.12.2.zip](https://github.com/jbredwards/Campfire-Mod/files/11183397/Campfire-v1.1.4-mc1.12.2.zip)

**Comment by SonicX8000** [comment]
Sorry for the late reply, I'll give it a go.

It looks like it's fixed... however... the game will crash if you close & reload a world while there's a placed brazier. It'll crash endlessly thus softlocking the world.

Crash:
[crash-2023-04-09_22.20.42-server.txt](https://github.com/jbredwards/Campfire-Mod/files/11187376/crash-2023-04-09_22.20.42-server.txt)

I first noticed this when I tried to load up an existing world with placed braziers due to showcasing the bug. So... if a world does NOT have a place brazier, it'll boot up fine... but if on exists... game crashes right away upon load.
